### PR TITLE
[#2599] Replace legacy northbound client with new interface in CLI.

### LINKED ISSUE #2599: Update Hono CLI to use new north bound client
Remove usage of legacy northbound client (`ApplicationClientFactory`) in _Hono CLI_ with the new client(`ApplicationClient`).

### PR DESCRIPTION
**Pull request by b-abel**
This fixes #2599.

### CODE REVIEW

File: cli/src/main/java/org/eclipse/hono/cli/app/AbstractApplicationClient.java [hunk 1]
```diff
@@ -33,10 +34,10 @@
     protected String deviceId;
     @Value(value = "${connection.retryInterval}")
     protected int connectionRetryInterval;
-    protected ApplicationClientFactory clientFactory;
+    protected ApplicationClient<? extends MessageContext> client;
 
     @Autowired
-    public final void setApplicationClientFactory(final ApplicationClientFactory factory) {
-        this.clientFactory = Objects.requireNonNull(factory);
+    public final void setApplicationClient(final ApplicationClient<? extends MessageContext> client) {
```
Comment (kaniyan) [inline cli/src/main/java/org/eclipse/hono/cli/app/AbstractApplicationClient.java:40]: It would be good to add  _JavaDoc_ for this _public_ _setter_.

Comment (b-abel) [reply cli/src/main/java/org/eclipse/hono/cli/app/AbstractApplicationClient.java:45]: Done.

**Comment by b-abel** [comment]
@kaniyan With the legacy `ApplicationClientFactory`, the client was closed after each command. My understanding is that this is no longer necessary with the current one. Is this right?

**Comment by kaniyan** [comment]
> @kaniyan With the legacy `ApplicationClientFactory`, the client was closed after each command. My understanding is that this is no longer necessary with the current one. Is this right?

@b-abel  Yes that's the current behavior of the `ApplicationClient`.

**Review by kaniyan — APPROVED** [review]
LGTM

File: cli/src/main/java/org/eclipse/hono/cli/app/AbstractApplicationClient.java [hunk 2]
```diff
@@ -33,10 +34,15 @@
...
     @Value(value = "${connection.retryInterval}")
     protected int connectionRetryInterval;
-    protected ApplicationClientFactory clientFactory;
+    protected ApplicationClient<? extends MessageContext> client;
 
+    /**
+     * Sets the application client to communicate with Hono's northbound API.
+     *
+     * @param client The client for the messaging system in use.
+     */
```
Comment (kaniyan) [inline cli/src/main/java/org/eclipse/hono/cli/app/AbstractApplicationClient.java:43]: How about also adding info on _NullPointerException_ to the _JavaDoc_?

**Pull request merged**